base/inference.jl

```diff
@@ -305,7 +305,8 @@ const workq = Vector{InferenceState}() # set of InferenceState objects that can
 
 #### helper functions ####
 
-@inline slot_id(s) = isa(s, SlotNumber) ? (s::SlotNumber).id : (s::TypedSlot).id # using a function to ensure we can infer this
+@inline slot_id(s::Slot) =
```

Comment:
type-inference should be good enough now that this won't cause a regression, but I'm not sure it's ideal to require it